src/testing.py

```diff
@@ -75,7 +75,10 @@ def setlimits():
   def __call__(self, test_file):
     """Execute a single test."""
     basename = os.path.basename(test_file)
-    outfile = self.outname_ctor(self.outdir, test_file) if self.outdir else ''
+    if self.outdir:
+      outfile = self.outname_ctor(self.outdir, test_file, self.extras)
+    else:
+      outfile = None
```

Comment:
Good point, in this context not having an outdir is probably a problem. And if we really want to dump output files to the current directory, we can pass `'.'` as the outdir.
So then, that's probably better encoded as `assert self.outdir, "Need to explicitly specify the output directory"` or so, rather than relying on `os.path.join` to probably blow up later.